data/console/test_java_transactional-server.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+# Variables Definition
+# --------------------
+PWD=$(/usr/bin/pwd)
+DIR="/var/tmp"
+LIST_INSTALLED_IBM_VERSIONS="$DIR/javas_ibm"
+LIST_INSTALLED_JDK_VERSIONS="$DIR/javas_jdk"
+LIST_INSTALLED_GCJ_VERSIONS="$DIR/javas.gcj"
+RPM_QUERY_JAVA="$DIR/zypper_se_java"
+LIST_ALL_INSTALLED_VERSIONS="$DIR/javas_installed"
+# --------------------
+# Test if there's any version of Java installed in the system
+is_java_installed() {
+    if [[ -s $LIST_ALL_INSTALLED_VERSIONS ]] ; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+
+# Find each (already) installed java version
+find_all_installed_java() {
+    # Construct temp files which are needed in order to
+    # determine how many java versions are installed along with their versions
+    # ignore javadoc as per issue https://progress.opensuse.org/issues/72070
+    rpm -qa | grep ^java- | grep -v javadoc > $RPM_QUERY_JAVA
+
+    # File that contains all the IBM Java installed in the system
+    grep "^java-" $RPM_QUERY_JAVA | grep "\-ibm" | awk -F 'ibm' '{print $1 "ibm"}' | sort -r | uniq > $LIST_INSTALLED_IBM_VERSIONS
+
+    # File that contains all the OpenJDK Java installed in the system
+    grep "^java-" $RPM_QUERY_JAVA | grep "openjdk" | awk -F 'openjdk' '{print $1 "openjdk"}' | sort -r | uniq > $LIST_INSTALLED_JDK_VERSIONS
+
+    # File that contains all the OpenJ9 Java installed in the system
+    grep "^java-" $RPM_QUERY_JAVA | grep "openj9" | awk -F 'openj9' '{print $1 "openj9"}' | sort -r | uniq >> $LIST_INSTALLED_JDK_VERSIONS
+
+    # File that contains all the GCC Java installed in the system
+    grep "^java-" $RPM_QUERY_JAVA | grep "gcj" | awk -F 'gcj' '{print $1 "gcj-compat"}' | sort -r | uniq > $LIST_INSTALLED_GCJ_VERSIONS
+
+    # Add JDK and IBM versions altogether, into one single file
+    cat $LIST_INSTALLED_JDK_VERSIONS $LIST_INSTALLED_IBM_VERSIONS $LIST_INSTALLED_GCJ_VERSIONS > $LIST_ALL_INSTALLED_VERSIONS
+}
+# ==== #
+# MAIN # Testing starts here ...
+# ==== #
+
+clear
+echo "----------------------------"
+echo "Find installed Java versions"
+echo "----------------------------"
+find_all_installed_java;
+if is_java_installed; then
+    cat $LIST_ALL_INSTALLED_VERSIONS
+else
+    echo "No Java versions found on the system"
+    exit 1
+fi
+echo "---------------------------------------------------------------------------------"
+echo "read only file system is not possible for some java tests. so skip further tests."
```

Comment:
What's the issue here? The full `test_java.sh` script should work fine.

Reply:
please my test: http://10.162.23.47/tests/8473#step/java/43

/usr/lib64 read-only file system.

Reply:
That looks like you could work around that by simply skipping the update-alternatives calls for the time being.
I'm not sure whether this is a bug in the java packages or in update-alternatives, maybe @laenion knows.

Reply:
Unfortunately that's a bug in `update-alternatives`. We have an internal Jira ticket for fixing this [jsc#SLE-9101], but there hasn't been any progress lately. So I guess the tests indeed have to be skipped for the time being. I'll try to ignite the ticket again to get a solution that just works on a transactional system.

Reply:
@laenion thanks for the info!